src/include/sof/audio/igo_nr/igo_nr_comp.h

```diff
@@ -0,0 +1,48 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2021 Intelligo Corporation. All rights reserved.
+ *
+ * Author: Fu-Yun TSUO <fy.tsuo@intelli-go.com>
+ */
+
+#ifndef __SOF_AUDIO_IGO_NR_CONFIG_H__
+#define __SOF_AUDIO_IGO_NR_CONFIG_H__
+
+#include <sof/platform.h>
+#include <sof/audio/audio_stream.h>
+#include <sof/audio/igo_nr/igo_lib.h>
+
+/* IGO_NR only support XCC HIFI3 */
+#if defined __XCC__ && XCHAL_HAVE_HIFI3 == 1
+#include <xtensa/config/core-isa.h>
+#else
+#error "No XCC/HIFI3 found. Cannot build IGO_NR module."
+#endif
+
+#define IGO_FRAME_SIZE (256)
+#define IGO_NR_IN_BUF_LENGTH (IGO_FRAME_SIZE)
+#define IGO_NR_OUT_BUF_LENGTH (IGO_FRAME_SIZE)
+#define IGO_MEM_SIZE         (40 * 1024)
+
+/* IGO_NR component private data */
+struct igo_nr_comp_data {
+	void *p_handle;
+	struct IgoLibInfo igo_lib_info;
+	struct IgoLibConfig igo_lib_config;
+	struct IgoStreamData igo_stream_data_in;
+	struct IgoStreamData igo_stream_data_ref;
+	struct IgoStreamData igo_stream_data_out;
+	struct comp_data_blob_handler *model_handler;
+	struct sof_igo_nr_config config;	    /**< blob data buffer */
+	int16_t in[IGO_NR_IN_BUF_LENGTH];	    /**< input samples buffer */
```

Comment:
this is fine for now, but its not future proof for future formats for sure

Reply:
this is library's audio format configuration.